### PR TITLE
Update package versions to latest

### PR DESCRIPTION
**Pull request by zachkont**
- Use LTS node - bumped minor version to reflect
- Migrate from tslint to eslint because it's [getting deprecated](https://medium.com/palantir/tslint-in-2019-1a144c2317a9)
- Fix linting

### CODE REVIEW

File: src/managers/init_manager.ts [hunk 1]
```diff
@@ -154,25 +154,15 @@ export default class InitManager {
     return this.addMissingUserFiles(dataPath);
   }
 
-  public static checkForKey(object: any, keyPath: string[], key: string): boolean {
```
Comment (TimJentzsch) [inline src/managers/init_manager.ts]: What happened to this function? Is it not used anywhere?

File: src/providers/provider_dota.ts [hunk 2]
```diff
@@ -27,8 +26,10 @@ export default class DotaProvider extends Provider {
       const newPatches = [];
 
       // Discard the old patches
+      // eslint-disable-next-line no-plusplus
```
Comment (TimJentzsch) [inline src/providers/provider_dota.ts]: Maybe we can actually use another type of loop here...

File: src/providers/provider_dota.ts [hunk 3]
```diff
@@ -27,8 +26,10 @@ export default class DotaProvider extends Provider {
...
       for (let i = 0; i < patchList.length && patchList[i] !== lastPatch; i++) {
         newPatches.push(patchList[i]);
+        this.logger.warn(patchList[i]);
```
Comment (TimJentzsch) [inline src/providers/provider_dota.ts]: Why is that logged? I'm pretty sure there is nothing to warn about here, but if there is, the logging message should be more helpful.

File: src/providers/provider_dota.ts [hunk 4]
```diff
@@ -71,6 +69,8 @@ export default class DotaProvider extends Provider {
     const patchList: string[] = [];
     const $ = pageDoc;
 
+    // This has to be a named function to set new `this` scope
```
Comment (TimJentzsch) [inline src/providers/provider_dota.ts]: I'd really like us to change this whole thing, the syntax is completely different than what we have everywhere else...

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gamefeeder",
-  "version": "0.8.5",
+  "version": "0.9.0",
```
[the remark this answers is not in the document]
Comment (zachkont) [reply package.json]: I doubt we'll merge something that bumps the minor version but in any case I'll have to revisit this once it's approved. If it conflicts with the other merge then it just serves as a reminded

File: src/commands/commands.ts [hunk 6]
```diff
@@ -297,36 +299,35 @@ const prefixCmd = new Command(
...
-      gameSubs: [],
-      id: channel.id,
-      prefix: newPrefix,
-    });
-    // Save the changes
-    subscribers[bot.name] = channels;
-    DataManager.setSubscriberData(subscribers);
-    return;
+
+    // Check if the channel is already registered
```
[the remark this answers is not in the document]
Comment (zachkont) [reply src/commands/commands.ts]: Whoops

File: src/managers/init_manager.ts (hunk 1 above)
Comment (zachkont) [reply src/managers/init_manager.ts]: Doesn't look like it was, also doesn't look like it was working properly

File: src/providers/provider_dota.ts (hunk 3 above)
Comment (zachkont) [reply src/providers/provider_dota.ts]: weird, I don't remember adding this... will remove

File: src/providers/provider_dota.ts (hunk 2 above)
Comment (zachkont) [reply src/providers/provider_dota.ts]: I didn't want to change the lastPatch check so I didn't touch it. We can create an issue and revisit later

File: src/commands/commands.ts [hunk 7]
```diff
@@ -297,36 +299,35 @@ const prefixCmd = new Command(
     const subscribers = DataManager.getSubscriberData();
     const channels = subscribers[bot.name];
 
-    // Check if the channel is already registered
```
[the remark this answers is not in the document]
Comment (zachkont) [reply src/commands/commands.ts]: Moved the comment

File: src/providers/provider_dota.ts (hunk 4 above)
Comment (zachkont) [reply src/providers/provider_dota.ts]: Let's create an issue about this and tackle it separately

File: src/bots/bot.ts [hunk 8]
```diff
@@ -179,37 +178,38 @@ export default abstract class BotClient {
...
+      return true;
     }
     return false;
   }
 
   /** Removes the data of all channels without write permissions. */
   public async removeChannelsWithoutWritePermissions() {
+    const user = await this.getUser();
     const channels = this.getBotChannels();
+    const permissionPromises = channels.map((channel) => this.getUserPermissions(user, channel));
```
[the remark this answers is not in the document]
Comment (zachkont) [reply src/bots/bot.ts]: done

File: src/bots/discord.ts [hunk 9]
```diff
@@ -244,29 +238,34 @@ export default class DiscordBot extends BotClient {
         const channels = this.getBotChannels();
 
         // Remove all channel data of that guild
-        for (const channel of channels) {
+        const removePromises = channels.map((channel) => {
```
[the remark this answers is not in the document]
Comment (zachkont) [reply src/bots/discord.ts]: done

File: src/providers/provider_dota.ts (hunk 2 above)
Comment (TimJentzsch) [reply src/providers/provider_dota.ts]: I moved this to issue #195.

File: src/providers/provider_dota.ts (hunk 4 above)
Comment (TimJentzsch) [reply src/providers/provider_dota.ts]: I moved this to issue #196.

File: src/managers/config_manager.ts [hunk 10]
```diff
@@ -51,25 +51,25 @@ export type reddit_config = {
...
+export type RedditUser = {
   /** The name of the reddit user. */
   name: string;
   /** The regex to filter the post tiles by. */
   titleFilter: string;
 };
 
 /** The reddit providers for a game. */
-export type reddit_providers = {
+export type RedditProvider = {
```
[the remark this answers is not in the document]
Comment (TimJentzsch) [reply src/managers/config_manager.ts]: I moved this to issue #197.

File: src/bots/discord.ts [hunk 11]
```diff
@@ -244,29 +238,34 @@ export default class DiscordBot extends BotClient {
...
       this.bot.on('channelDelete', async (discordChannel) => {
         const channels = this.getBotChannels();
 
-        // Search for the deleted channel
-        for (const channel of channels) {
-          if (channel.id === discordChannel.id) {
-            await this.onRemoved(channel);
-          }
+        // Search for the channel
+        const channel = channels.find((ch) => discordChannel.id === ch.id);
```
[the remark this answers is not in the document]
Comment (TimJentzsch) [reply src/bots/discord.ts]: Every channel should only appear once, so this should be fine.

File: src/bots/telegram.ts [hunk 12]
```diff
@@ -243,11 +245,9 @@ export default class TelegramBot extends BotClient {
           const channels = this.getBotChannels();
           const channelID = message.chat.id.toString();
           // Search for the channel
-          for (const channel of channels) {
-            if (channelID === channel.id) {
-              // Remove channel data
-              await this.onRemoved(channel);
-            }
+          const channel = channels.find((ch) => channelID === ch.id);
```
[the remark this answers is not in the document]
Comment (TimJentzsch) [reply src/bots/telegram.ts]: Every chat should only appear once, so this should be fine.

**Pull request merged**